pkg/controller/customdomain/customdomain_controller.go

```diff
@@ -133,532 +131,109 @@ func (r *ReconcileCustomDomain) Reconcile(request reconcile.Request) (reconcile.
 	}
```

Comment:
[Latent] For idempotence, could make the above (finalizer addition) return on success too.